dev/RadioButtons/RadioButtons.cpp

```diff
@@ -7,136 +7,634 @@
 #include "Vector.h"
 #include "RuntimeProfiler.h"
 #include "ResourceAccessor.h"
+#include "RadioButtonsTestHooks.h"
 
 RadioButtons::RadioButtons()
 {
     __RP_Marker_ClassById(RuntimeProfiler::ProfId_RadioButtons);
 
-    auto items = winrt::make<Vector<winrt::IInspectable>>();
+    m_selectionModel = winrt::SelectionModel{};
+    m_selectionModel.SingleSelect(true);
+    m_selectionModel.Source(GetItemsSource());
+    m_selectionChangedRevoker = m_selectionModel.SelectionChanged(winrt::auto_revoke, { this, &RadioButtons::OnSelectionChanged });
```

Comment:
During our discussion, you mentioned that SelectionChanged is firing more often than required. Can you please file a bug for it. If the selection has not changed, we should not be firing the event.

Reply:
#1642

---
In reply to: [347573403](https://github.com/microsoft/microsoft-ui-xaml/pull/1623#discussion_r347573403) [](ancestors = 347573403)